addons/website_sale_stock/models/product_template.py

```diff
@@ -1,20 +1,34 @@
 # -*- coding: utf-8 -*-
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
-from odoo import fields, models, api
-
+from odoo import fields, models, api, _
+from odoo.exceptions import ValidationError
 
 class ProductTemplate(models.Model):
     _inherit = 'product.template'
 
-    inventory_availability = fields.Selection([
-        ('never', 'Sell regardless of inventory'),
-        ('always', 'Show inventory on website and prevent sales if not enough stock'),
-        ('threshold', 'Show inventory below a threshold and prevent sales if not enough stock'),
-        ('custom', 'Show product-specific notifications'),
-    ], string='Inventory Availability', help='Adds an inventory availability status on the web product page.', default='never')
-    available_threshold = fields.Float(string='Availability Threshold', default=5.0)
-    custom_message = fields.Text(string='Custom Message', default='', translate=True)
+    @api.model
+    def _get_config_value(self, key):
+        return self.env['res.config.settings'].get_values().get(key)
+
+    allow_to_order = fields.Selection([
+        ('available', 'Allow the user to order only if there is enough quantity on hand for the product'),
+        ('always', 'Allow the user to order even if there is no quantity on hand for the product'),
+    ], string='Allow to order', default=lambda self: self._get_config_value('allow_to_order'))
+    available_threshold = fields.Float(string='Availability Threshold', default=lambda self: self._get_config_value('available_threshold'))
+    available_message = fields.Html(string="In Stock", default=lambda self: self._get_config_value('available_message'))
+    out_of_stock_message = fields.Html(string="Out of Stock", default=lambda self: self._get_config_value('out_of_stock_message'))
+    below_threshold_message = fields.Html(string="Below Threshold", default=lambda self: self._get_config_value('below_threshold_message'))
+
+    @api.constrains('available_message', 'out_of_stock_message', 'below_threshold_message')
+    def _check_stock_message(self):
+        for record in self:
+            try:
+                record.available_message.format(qty='', unit='')
+                record.out_of_stock_message.format(qty='', unit='')
+                record.below_threshold_message.format(qty='', unit='')
+            except KeyError:
+                raise ValidationError(_("You did not use correctly one of the two variables: {qty} or {unit}"))
```

Comment:
I think %s instead of {} when inside translation, so translator doesn't try to translate the variable name?